src/test/resources/recipes/sample-template-with-all-fields.json

```diff
@@ -0,0 +1,64 @@
+{
+  "RecipeFormatVersion": "2020-01-25",
+  "ComponentName": "FooTemplate",
+  "ComponentDescription": "Test template for Greengrass packages",
+  "ComponentPublisher": "Me",
+  "ComponentVersion": "1.0.0",
+  "ComponentType": "aws.greengrass.template",
+  "ComponentConfiguration": {
+    "ParameterSchema": {
```

Comment:
why is this under component configuration anyhow? Schema isn't config really.... I think this should maybe be a different top-level key.

Reply:
Semantically, we need a `Parameters` field for `ComponentConfiguration` for parameter files, since they can't be modified like normal config K/Vs but also are config values. So either we put both `Parameters` and `ParameterSchema` as top-level keys or we stick both in `ComponentConfiguration`

Reply:
Well I'd want the schema taken out from the user input completely